### PR TITLE
[ecotouch] Add water and heating power limit channels

### PR DESCRIPTION
**Pull request by troem**
# [ecotouch] Limit power for heatpump
Addition of 4 new channels for limiting the power of the heatpump
Percent Water Power limit Min
Percent Water Power limit Max
Percent Heating Power limit Min
Percent Heating Power limit Min

### CODE REVIEW

File: bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/i18n/ecotouch_de.properties [hunk 1]
```diff
@@ -179,6 +179,10 @@ channel-type.ecotouch.temperature_water_set.label = Sollwert Warmwasserspeicher
 channel-type.ecotouch.temperature_water_set.description = Geforderte Temperatur im Warmwasserspeicher
 channel-type.ecotouch.temperature_water_set2.label = BMS Sollwert Warmwasserspeicher
 channel-type.ecotouch.temperature_water_set2.description = Solltemperatur im Warmwasserspeicher vom BMS
+channel-type.ecotouch.percent_water_limit_min.description = Warmwasser Leistungsbegrenzung Min Ausgang
```
Comment (lsiepel) [inline bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/i18n/ecotouch_de.properties:182]: Please revert changes to this file, translations should be made by crowdin.

File: bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/i18n/ecotouch.properties [hunk 2]
```diff
@@ -179,6 +179,10 @@ channel-type.ecotouch.temperature_water_set.label = Water Set
 channel-type.ecotouch.temperature_water_set.description = Temperature Water Setpoint
 channel-type.ecotouch.temperature_water_set2.label = Water Set BMS
 channel-type.ecotouch.temperature_water_set2.description = Temperature Water Setpoint BMS
```
Comment (lsiepel) [inline bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/i18n/ecotouch.properties:181]: New labels are midding to this file, can you re-run the i18n plugin?

File: bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml [hunk 3]
```diff
@@ -888,7 +888,26 @@
 		<category>Temperature</category>
 		<state readOnly="true" pattern="%.1f %unit%"/>
 	</channel-type>
-
+	<channel-type id="percent_water_limit_min" advanced="true">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Percent Water limit min</label>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml:893]: Please adjust label to 2-3 words, < 23 characters. and capitalized.

File: bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml [hunk 4]
```diff
@@ -888,7 +888,26 @@
...
+		<state readOnly="false" pattern="%.1f %unit%"/>
+	</channel-type>
+	<channel-type id="percent_water_limit_max" advanced="true">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Percent Water limit max</label>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml:898]: Please adjust label to 2-3 words, < 23 characters. and capitalized.

File: bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml [hunk 5]
```diff
@@ -888,7 +888,26 @@
...
+		<state readOnly="false" pattern="%.1f %unit%"/>
+	</channel-type>
+	<channel-type id="percent_heating_limit_min" advanced="true">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Percent Heating limit min</label>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml:903]: Please adjust label to 2-3 words, < 23 characters. and capitalized.

File: bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml [hunk 6]
```diff
@@ -888,7 +888,26 @@
...
+		<state readOnly="false" pattern="%.1f %unit%"/>
+	</channel-type>
+	<channel-type id="percent_heating_limit_max" advanced="true">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Percent Heating limit max</label>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml:908]: Please adjust label to 2-3 words, < 23 characters. and capitalized.

**Review by lsiepel — CHANGES_REQUESTED** [review]
Left some comments. Please also provide Thing upgrade instructions: https://www.openhab.org/docs/developer/bindings/thing-xml.html#updating-thing-types

**Comment by troem** [comment]
> Left some comments. Please also provide Thing upgrade instructions: https://www.openhab.org/docs/developer/bindings/thing-xml.html#updating-thing-types
> 
> Edit: it also needs spotless and the commits needs to be signed off.

OK - thank you for all your comments - I tried to do all my homework now and followed your instructions. This is my very first pull request and I am sorry heaving not read all documents carefully enough. Thank you for your time!

File: bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml [hunk 7]
```diff
@@ -888,7 +897,26 @@
 		<category>Temperature</category>
 		<state readOnly="true" pattern="%.1f %unit%"/>
 	</channel-type>
-
+	<channel-type id="percent_water_limit_min" advanced="true">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Water Powerlimit min</label>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml:902]:
```suggestion
		<label>Water Powerlimit Min</label>
```

File: bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml [hunk 8]
```diff
@@ -888,7 +897,26 @@
...
+		<state readOnly="false" pattern="%.1f %unit%"/>
+	</channel-type>
+	<channel-type id="percent_water_limit_max" advanced="true">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Water Powerlimit max</label>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml:907]:
```suggestion
		<label>Water Powerlimit Max</label>
```

File: bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml [hunk 9]
```diff
@@ -888,7 +897,26 @@
...
+		<state readOnly="false" pattern="%.1f %unit%"/>
+	</channel-type>
+	<channel-type id="percent_heating_limit_min" advanced="true">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Heating Powerlimit min</label>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml:912]:
```suggestion
		<label>Heating Powerlimit Min</label>
```

File: bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml [hunk 10]
```diff
@@ -888,7 +897,26 @@
...
+		<state readOnly="false" pattern="%.1f %unit%"/>
+	</channel-type>
+	<channel-type id="percent_heating_limit_max" advanced="true">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Heating Powerlimit max</label>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.ecotouch/src/main/resources/OH-INF/thing/thing-types.xml:917]:
```suggestion
		<label>Heating Powerlimit Max</label>
```

**Review by lsiepel — CHANGES_REQUESTED** [review]
Some final comments.

After the labels have been adjusted also run the i18n plugin again to generate the properties file. 

Regarding the thing-type instructions one last bit is missing.  Add this to both Thing's. Doc here: https://www.openhab.org/docs/developer/bindings/thing-xml.html#updating-thing-types
```
<properties>
    <property name="thingTypeVersion">1</property>
  </properties>
```

**Comment by troem** [comment]
> Some final comments.
> 
> After the labels have been adjusted also run the i18n plugin again to generate the properties file.
> 
> Regarding the thing-type instructions one last bit is missing. Add this to both Thing's. Doc here: https://www.openhab.org/docs/developer/bindings/thing-xml.html#updating-thing-types
> 
> ```
> <properties>
>     <property name="thingTypeVersion">1</property>
>   </properties>
> ```



> Some final comments.
> 
> After the labels have been adjusted also run the i18n plugin again to generate the properties file.
> 
> Regarding the thing-type instructions one last bit is missing. Add this to both Thing's. Doc here: https://www.openhab.org/docs/developer/bindings/thing-xml.html#updating-thing-types
> 
> ```
> <properties>
>     <property name="thingTypeVersion">1</property>
>   </properties>
> ```

Done with cf01ee5514154d08ba1afd193800dac0bbe9d9eb 
Thank you for your review

**Comment by lsiepel** [comment]
Did you close this on purpose?

**Comment by troem** [comment]
> Did you close this on purpose?

So I thought I did all my homework, obviously I created a mess, I am sorry. I saw the "Closed with unmerged commits" but I did not find a way to merge / to solve this situation ... could you give me a hint, please?

**Review by lsiepel — APPROVED** [review]
Thanks, LGTM

**Pull request merged**